src/track/beatmap.cpp

```diff
@@ -269,7 +269,7 @@ QString BeatMap::getSubVersion() const {
 }
 
 bool BeatMap::isValid() const {
-    return m_sampleRate.isValid() && m_beats.size() > 0;
+    return m_sampleRate.isValid() && m_beats.size() > 1;
```

Comment:
Why is a beatgrid invalid if it has one beat?

Reply:
Because it has no tempo.

Reply:
I suppose this is worth a comment. Better: Introduce a constant `constexpr int kMinNumberOfBeats = 2` and place the comment there.

Reply:
Class invariant?

`DEBUG_ASSERT(m_beats.isEmpty() || m_beats.size() >= kMinNumberOfBeats);`